ibm/service/directlink/resource_ibm_dl_gateway_action.go

```diff
@@ -0,0 +1,1187 @@
+		Create:   resourceIBMdlGatewayCreateAction,
+		Read:     resourceIBMdlGatewayActionRead,
+		Update:   resourceIBMdlGatewayActionUpdate,
+		Delete:   resourceIBMdlGatewayActionDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
+				return flex.ResourceTagsCustomizeDiff(diff)
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			dlGatewayId: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Direct Link gateway identifier",
+			},
+			dlCustomerAction: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlCustomerAction),
+				Description:  "customer action on provider call",
+			},
+			dlAuthenticationKey: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "BGP MD5 authentication key",
+			},
+			dlExportRouteFilters: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "List Export Route Filters for a Direct Link gateway",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						dlExportRouteFilter: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Export route Filter identifier",
+						},
+						dlAction: {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.InvokeValidator("ibm_dl_gateway_action", dlAction),
+							Description:  "Determines whether the  routes that match the prefix-set will be permit or deny",
+						},
+						dlBefore: {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Identifier of the next route filter to be considered",
+						},
+						dlCreatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date and time of the export route filter was created",
+						},
+						dlGe: {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "The minimum matching length of the prefix-set",
+						},
+						dlLe: {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "The maximum matching length of the prefix-set",
+						},
+						dlPrefix: {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "IP prefix representing an address and mask length of the prefix-set",
+						},
+						dlUpdatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date and time of the export route filter was last updated",
+						},
+					},
+				},
+			},
+			dlImportRouteFilters: {
```

Comment:
if any change in filters its a force new?

Reply:
no, it will be updated.